### ISSUE TITLE
Animation does not refresh correctly

### ISSUE THREAD

**Issue by kmungantiwar** [highfestiva/finplot#222]
Hello,

The animation adds 2 candles with each refresh whereas it should have added only one. Can you help with the same.

### Code to reproduce

```python
import finplot as fplt
import yfinance

filecounter = 0
df = yfinance.download('RELIANCE.NS', period='2mo')
df = df.drop(columns='Adj Close')

def animate():
    global filecounter, ax, df
    try:
        fplt.set_x_pos(df.index[int(filecounter)], df.index[int(filecounter+20)], ax=ax)
        filecounter += 1
    except:
        fplt.close()

ax = fplt.create_plot('Things move', init_zoom_periods=20, maximize=False)
fplt.candlestick_ochl(df[['Open', 'Close', 'High', 'Low']])

fplt.timer_callback(animate, 1)
fplt.show()
```

### Describe the bug
The animation, with each refresh should have added one new candle to the right but instead adds two candles. have kept the refresh rate low so that it can be clearly seen

**Comment by highfestiva** [comment]
Interesting bug, investigating...

**Comment by highfestiva** [comment]
Cool, learnt something new today. The reason was that Python (by default, think it was at least in the past possible to change) rounds to closest even number. I.e. 2.5 rounds to 2. Fixed in 1190e8af6e181c4faf0d4be012eca9102ed7fd71.

**Comment by kmungantiwar** [comment]
Thank you for helping out!
